src/Castle.Core/DynamicProxy/Internal/AttributeUtil.cs

```diff
@@ -204,7 +224,7 @@ public static IEnumerable<CustomAttributeInfo> GetNonInheritableAttributes(this
 				var attributeType = attribute.AttributeType;
 #endif
 
-				if (ShouldSkipAttributeReplication(attributeType))
+				if (ShouldSkipAttributeReplication(attributeType, ignoreInheritance))
```

Comment:
Could you use `(parameter.Member as ConstructorInfo) != null` rather than those new methods.

Not sure why I wrote that inheritance could be the problem in the issue, because obviously constructors are not part of an class inheritance contract and must be redefined, but maybe I was thinking it could be this code excluding them because they are marked inheritable 😉.

Reply:
I don't quite understand. Could you please clarify which new methods you'd like to avoid and where that condition should go?

* `ShouldSkipAttributeReplication` needs an additional parameter so that it knows whether or not to bypass the check of `AttributeUsageAttribute.Inherited`. It's only given a `Type` but no `ParameterInfo`, after all, so a `parameter.Member is ConstructorInfo` check cannot go inside this method.

* `parameter.Member is ConstructorInfo` can however be placed in the original `GetNonInheritableAttributes` to avoid the two new forwarding methods `GetCustomAttributesToBeReproduced` and `GetCustomAttributesToBeReproducedOnProxyCtor`. However, doing so means that `GetNonInheritableAttributes` will in some circumstances return inheritable attributes, which goes directly against its name and *might* seem like a bug to someone coming across that code for the first time.

Reply:
Yep, I would like to avoid the 2 new forwarding methods as they seem unnecessary. I was thinking you could determine if you are handling a constructor in `GetNonInheritableAttributes`, pretty much thinking:

```csharp
bool ignoreInheritance = (parameter.Member is ConstructorInfo);
```

Since constructors in .NET are not inherited (not that you'd want them to be), I don't see any problem returning all attributes via this named method because no constructor parameters can ever be inherited.

Reply:
Ah, gotcha. Makes sense now. Done.